File: clients/Python/examples/MqClientPoolExample.py
import sys
sys.path.append("../")
from zbus import MqClientPool, Message

pool = MqClientPool("localhost:15555")
def on_connected(server_info):
    print(server_info)

pool.on_connected = on_connected
pool.start()


client = pool.borrow_client() 

msg = Message()
msg.topic = 'hong'
msg.body = 'hello world'

client.produce(msg)
res = client.consume('hong') 
print (res) 

pool.return_client(client)
pool.close()

File: clients/CPP/examples/MqClientPoolExample.cpp
#include "MqClient.h"  


int main_MqClientPool(int argc, char* argv[]) {  
	Logger::configDefaultLogger(0, LOG_DEBUG); 
	Logger* log = Logger::getLogger();

	{
		MqClientPool pool("localhost:15555"); 
		MqClient* client = pool.borrowClient(); 
		client->queryTracker(); 
		pool.returnClient(client);
	}

	system("pause");
	return 0;
}